viewing_party/party.py

```diff
@@ -0,0 +1,264 @@
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    """
+    Takes in parameter user_data,
+    returns the most watched genre among the movies.
+    """
+    most_watched = user_data["watched"]
+    genres = []
+    if len(most_watched) == 0:
+        return None
+    for movie in most_watched:
+        genres.append(movie["genre"])
+    # consider the big o notation of importing statistics, mode    
+    return statistics.mode(genres)
+
+
+# Wave 3
+def get_unique_watched(user_data):
+    """
+    Takes one parameter user_data,
+    returns list of dictionaries unique_watched, 
+    representing movies that the user has watched,
+    but none of their friends have watched.
+    """
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    
+    friends_titles = []
+    for friend in friends_watched:
+        for title in friend["watched"]:
+            friends_titles.append(title['title'])
+
+    unique_watched = []
+
+    for title in user_watched:
+        if title["title"] not in friends_titles:
+            unique_watched.append(title)
+
+    return unique_watched
+    
+    
+def get_friends_unique_watched(user_data):
+    """
+    Takes in one parameter user_data,
+    returns list of dictionaries friends_unique_watched,
+    representing movies at least one friend has watched,
+    but the user has not watched. 
+    """
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    
+    user_movies = [movie["title"] for movie in user_watched]
+    friends_unique_watched = []
+
+    for friend in friends_watched:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_movies:
+                if movie not in friends_unique_watched:
+                    friends_unique_watched.append(movie)
+
+    return friends_unique_watched
+
+
+# Wave 4
+def user_has_watched(movie, watched_movies):
+    """
+    Takes two parameters, movie and watched_movies.
+    Determines if user has watched a movie,
+    returns a Boolean. 
+    """
+    if movie in watched_movies:
+        return True
+    else:
+        return False
+
+
+def in_friends_watchlist(movie, watched_by_friends):
+    """
+    Takes two parameters, movie and watched_by_friends.
+    Determines whether at least one friend has watched the movie.
+    Returns a Boolean.
+    """
+    if movie in watched_by_friends:
+        return True
+    else:
+        return False
+
+
+def in_users_subscriptions(host, subscriptions):
+    """
+    Takes two parameters, host and subscriptions.
+    Determines whether the movie host is a service subscription.
+    Returns a Boolean.
+    """
+    if host in subscriptions:
+        return True
+    else:
+        return False
+
+
+def get_friends_movies(user_data):
+    """
+    Takes in one parameter user_data.
+    Determines which movies friends has watched.
+    Returns list of dictionaries, movies.
+    """
+    titles = set()
```

Comment:
oh interesting! any reason for using a set here? it doesn't look like you are comparing it to another set, so is it making sure no duplicate titles are added? cool